documentation/src/main/asciidoc/introduction/Interacting.adoc

```diff
@@ -995,7 +997,7 @@ The `getResultCount()` method is useful for displaying the number of pages of re
 SelectionQuery<Book> query =
         session.createSelectionQuery("from Book where title like ?1 order by title", Book.class)
             .setParameter(1, titlePattern);
-long pages = query.getResultCount() / MAX_RESULTS;
+long pages = (long) Math.ceil(query.getResultCount() * 1.0 / MAX_RESULTS);
```

Comment:
Why is this change necessary?

Reply:
suppose the total number is 50 and max result per page is 20, then the page number should be 3 (`Math.ceil(50.0 / 20)`, the last page has 10 results) instead of 2 (`50 / 20` in Java).

Not sure my implementation is best though.

Reply:
I'm just not sure I wanna be blinding the reader with that level of detail here.

Reply:
Perhaps the following is more immediately readable:

```java
long results = query.getResultCount();
long pages = results / MAX_RESULTS + (results % MAX_RESULTS == 0 ? 0 : 1);
```

Reply:
changed as suggested